crates/red_knot_python_semantic/src/types.rs

```diff
@@ -328,8 +328,7 @@ impl<'db> Type<'db> {
         match self {
             Type::Function(function_type) => function_type.returns(db).or(Some(Type::Unknown)),
 
-            // TODO: handle class constructors
-            Type::Class(_class_ty) => Some(Type::Unknown),
+            Type::Class(_class_ty) => Some(self.instance()),
```

Comment:
```suggestion
            // TODO annotated return type on `__new__` or metaclass `__call__`
            Type::Class(_class_ty) => Some(self.instance()),
```

Reply:
On second look I made one other small change here. We don't need to match again in `self.instance` when we've already matched once and know we have a `Type::Class` -- the definition of `instance()` for `Type::Class` is quite obvious and short, it's fine to repeat it here. The main purpose of `instance()` is to abstract across `Type` variants, which we don't need here.